openvasd/openvasd.c

```diff
@@ -914,6 +913,7 @@ openvasd_get_scan_status (openvasd_connector_t conn)
     }
   else
     {
+      response = g_malloc0 (sizeof (struct openvasd_response));
```

Comment:
It looks like there are are cases that need this, eg `openvasd_stop_scan`.

And for example `openvasd_get_scan_results` is leaking the result because it allocates it at the top, then `openvasd_send_request` assigns over that.

Happy to do a PR for these.